fastapi_security/basic.py

```diff
@@ -1,12 +1,29 @@
 import secrets
+from base64 import urlsafe_b64encode
 from typing import Dict, Iterable, List, Union
 
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.hashes import SHA512, Hash
+
 from fastapi.security.http import HTTPBasicCredentials
 
-__all__ = ("HTTPBasicCredentials",)
+__all__ = ("HTTPBasicCredentials", "generate_digest")
+
+
+from pydantic import BaseModel
+
+
+class HTTPBasicCredentialsDigest(BaseModel):
+    username: str
+    digest: str
+
```

Comment:
Maybe reuse HTTPBasicCredentials instead?
```suggestion
```

Reply:
I'm not sure I understand the suggestion here.

It is an import from `fastapi.security.http`, and it is used in fastapi in the context where `username` and `password` fields make sense. How would you reuse `HTTPBasicCredentials` to store a password digest instead?

Reply:
Ok, is your idea is to store the digest in the `password` field of `HTTPBasicCredentials`? It can be made to work, it's slightly confusing, but it's alright.